### ISSUE TITLE
`random()` has slightly biased distribution

### ISSUE THREAD

**Issue by dimitri-xyz** [MikeMcl/decimal.js#52]
This is a fantastic library! Thanks! One small issue.

The normalization of the exponent done at [the end of `random()`](https://github.com/MikeMcl/decimal.js/blob/3566299686993864f1a795954f189cdfbf8e4ff9/decimal.js#L4637-L4641) normalizes the exponent but does not do the corresponding shift of the "mantissa digits" (as was done in the [line above](https://github.com/MikeMcl/decimal.js/blob/3566299686993864f1a795954f189cdfbf8e4ff9/decimal.js#L4635) with `rd.shift()`).

This means that the numbers produced that have leading zeros will not have a normalized "mantissa" and as a consequece will be biased towards being smaller than they should be.

If normalization is not required than [these lines](https://github.com/MikeMcl/decimal.js/blob/3566299686993864f1a795954f189cdfbf8e4ff9/decimal.js#L4637-L4641) should be removed. Otherwise, we must also shift the mantissa (i.e. `rd`) to the left by `LOG_BASE - k` digits after normalizing the exponent.

**Comment by MikeMcl** [comment]
Thanks for the feedback, but I don't think there is any bias.

The lines you suggest should be removed are essential to calculate the exponent of the number that has been produced.

For example, if `Decimal.precision = 14` and the 14 random digits produced were `00045671234567` then `rd` would be `[4567, 1234567]` and the resulting random numnber would be `0.00045671234567`, which is stored in floating point form, i.e. `4.5671234567e-4`.

Please give example an example of how you think the bias occurs.

**Comment by dimitri-xyz** [comment]
My apologies! I clearly don't know the library well and misunderstood how you normalize the representation.

I assumed you also normalized the first element of `rd` to always contain 7 digits. In other words,`rd = [4567, 1234567]` and `e=-4` would be interpreted as `[0004567, 1234567]` (note the zeros on the left) and `e=-4` representing 0.00045671234567 x 10ˆ(-4) = 0.000000045671234567 which is 4.5671234567e-8 (generating the bias I described).

In my mistaken interpretation, we would represent 0.00045671234567 with `rd = [4567123, 4567000]` and `e=-4`. That's why I talked about shifting the mantissa.
